src/pages/donate.tsx

```diff
@@ -20,29 +33,71 @@ type DonationData = {
 const CREATE_DONATION_MUTATION = gql`
   mutation DonatePageCreate($input: CreateTrackableInput!) {
     createUnownedTrackable(input: $input) {
-      code
+      trackable {
+        did
+        ownerKey {
+          publicKey
+          privateKey
+        }
+      }
+    }
+  }
+`
+
+const UPDATE_DONATION_MUTATION = gql`
+  mutation UpdateDonation($input: AddUpdateInput!) {
+    addUpdate(input: $input) {
+      did
+      message
+      metadata
     }
   }
 `
 
 export function DonatePage() {
   const [createDonation, { error: createError }] = useMutation(CREATE_DONATION_MUTATION)
-  const { handleSubmit, errors, setError, register, formState } = useForm<DonationData>();
+  const [updateDonation, { error: updateError }] = useMutation(UPDATE_DONATION_MUTATION)
+
+  const { handleSubmit, errors, register } = useForm<DonationData>();
+
+  const history = useHistory()
 
   async function onSubmit({pickupAddr,instructions}:DonationData) {
     setSubmitLoading(true)
     log("submitted:", pickupAddr, instructions);
 
-    // first create the trackable
-    // TODO: Add the image once that's working
-    await createDonation({
+    const result = await createDonation({
       variables: {input: {name: "donation"}}
     })
 
-    // TODO: add the location in an update
-    // How do we get the new trackable?
+    const payload:CreateTrackablePayload = result.data.createUnownedTrackable
+    const donation:Trackable = payload.trackable!
+
+    log("createDonation result:", donation)
 
-    setSubmitLoading(false)
+    let metadata:MetadataEntry[] = []
+
+    metadata.push({key: "location", value: pickupAddr})
+
+    // TODO: Add the image once that's working
+    //metadata.push(key: "image", value: skylink)
+
+    log("adding metadata:", metadata)
+
+    await updateDonation({
```

Comment:
whats the purpose of doing a create then update back to back?

Reply:
None really. It was just how the app was already structured. Location was an update metadata thing.

Reply:
if we can bake this all into create, does that remove the need for the `ownerKey` pass through?

Reply:
Mayyyyybe, but I think you'd need it later on anyway. However, your change in #12 to make it owned by the drivers' / chapter chaintree I think _does_ obviate the need for it. Right?

Reply:
I'm going to rebase this on latest master since #12 is merged and see what shakes loose.

Reply:
@brandonwestcott OK rebased, and all key passing is gone. I'm just using `drivers.key()` for all updates now.